pocs/camera/simulator.py

```diff
@@ -83,26 +89,31 @@ def take_observation(self, observation, headers=None, **kwargs):
 
         return camera_event
 
-    def take_exposure(self, seconds=1.0 * u.second, filename=None):
+    def take_exposure(self, seconds=1.0 * u.second, filename=None, dark=False, blocking=False):
         """ Take an exposure for given number of seconds """
+        assert self.is_connected, self.logger.error("Camera must be connected for take_exposure!")
 
         assert filename is not None, self.logger.warning("Must pass filename for take_exposure")
 
         if isinstance(seconds, u.Quantity):
+            seconds = seconds.to(u.second)
             seconds = seconds.value
 
-        self.logger.debug('Taking {} second exposure on {}'.format(seconds, self.name))
+        self.logger.debug('Taking {} second exposure on {}: {}'.format(seconds, self.name, filename))
 
-        # Simulator just sleeps
-        run_cmd = ["sleep", str(seconds)]
+        # Set up a Timer that will wait for the duration of the exposure then copy a dummy FITS file
+        # to the specified path and adjust the headers according to the exposure time, type.
+        start_time = Time.now()
+        exposure_event = Event()
+        exposure_thread = Timer(interval=seconds,
+                                function=self._fake_exposure,
+                                args=[seconds, start_time, filename, exposure_event, dark])
+        exposure_thread.start()
 
-        # Send command to camera
-        try:
-            proc = subprocess.Popen(run_cmd, stdout=subprocess.PIPE, stderr=subprocess.STDOUT, universal_newlines=True)
-        except error.InvalidCommand as e:
-            self.logger.warning(e)
+        if blocking:
```

Comment:
Does the blocking need to happen within the `take_exposure`? For other exposures the "blocking" happens within the state machine itself, while `POCS` is in the `observing` state (or `pointing`). It's also not a true block, but instead checks for interrupt messages to the system as a whole. 

See https://github.com/panoptes/POCS/blob/develop/pocs/state/states/default/observing.py#L18

Can we have the user of `take_exposure` control the blocking or is there a reason to have it here?